bridges/snowbridge/runtime/test-common/src/lib.rs

```diff
@@ -193,9 +195,104 @@ pub fn send_transfer_token_message_success<Runtime, XcmConfig>(
+				DigestItem::Consensus(_, _) => panic!("is Consensus"),
+				DigestItem::Seal(_, _) => panic!("is Seal"),
+				DigestItem::Other(veccc) => {
+					assert_eq!(veccc[0], 0);
+				},
+				DigestItem::RuntimeEnvironmentUpdated => panic!("is RuntimeEnvironmentUpdated"),
+			}
+		});
+}
+
+pub fn test_pallet_order_works<Runtime, XcmConfig, AllPalletsWithoutSystem>(
+	collator_session_key: CollatorSessionKeys<Runtime>,
+	runtime_para_id: u32,
+	assethub_parachain_id: u32,
+	weth_contract_address: H160,
+	destination_address: H160,
+	fee_amount: u128,
+	snowbridge_pallet_outbound_queue: Box<
+		dyn Fn(Vec<u8>) -> Option<snowbridge_pallet_outbound_queue::Event<Runtime>>,
+	>,
+) where
+	Runtime: frame_system::Config
+		+ pallet_balances::Config
+		+ pallet_session::Config
+		+ pallet_xcm::Config
+		+ parachain_info::Config
+		+ pallet_collator_selection::Config
+		+ pallet_message_queue::Config
+		+ cumulus_pallet_parachain_system::Config
+		+ snowbridge_pallet_outbound_queue::Config
+		+ snowbridge_pallet_system::Config
+		+ pallet_timestamp::Config,
+	XcmConfig: xcm_executor::Config,
+	AllPalletsWithoutSystem:
+		OnInitialize<BlockNumberFor<Runtime>> + OnFinalize<BlockNumberFor<Runtime>>,
+	ValidatorIdOf<Runtime>: From<AccountIdOf<Runtime>>,
+	<Runtime as frame_system::Config>::AccountId: From<sp_runtime::AccountId32> + AsRef<[u8]>,
+{
+	ExtBuilder::<Runtime>::default()
+		.with_collators(collator_session_key.collators())
+		.with_session_keys(collator_session_key.session_keys())
+		.with_para_id(runtime_para_id.into())
+		.with_tracing()
+		.build()
+		.execute_with(|| {
+			<snowbridge_pallet_system::Pallet<Runtime>>::initialize(
+				runtime_para_id.into(),
+				assethub_parachain_id.into(),
+			)
+			.unwrap();
+
+			// fund asset hub sovereign account enough so it can pay fees
+			initial_fund::<Runtime>(assethub_parachain_id, 5_000_000_000_000);
+
+			let outcome = send_transfer_token_message::<Runtime, XcmConfig>(
+				assethub_parachain_id,
+				weth_contract_address,
+				destination_address,
+				fee_amount,
+			);
+
+			assert_ok!(outcome.ensure_complete());
+
+			// check events
+			let mut events = <frame_system::Pallet<Runtime>>::events()
+				.into_iter()
+				.filter_map(|e| snowbridge_pallet_outbound_queue(e.event.encode()));
+			assert!(events.any(|e| matches!(
+				e,
+				snowbridge_pallet_outbound_queue::Event::MessageQueued { .. }
+			)));
+
+			let next_block_number: U256 = <frame_system::Pallet<Runtime>>::block_number()
+				.saturating_add(BlockNumberFor::<Runtime>::from(1u32))
+				.into();
+
+			let included_head =
+				RuntimeHelper::<Runtime, AllPalletsWithoutSystem>::run_to_block_with_finalize(
+					next_block_number.as_u32(),
+				);
+			let digest = included_head.digest();
+			let digest_items = digest.logs();
+
+			let mut found_outbound_digest = false;
+			for digest_item in digest_items {
+				match digest_item {
+					DigestItem::Other(_) => found_outbound_digest = true,
```

Comment:
So we assert that the message is queued on a block and then the DigestItem exists in the next block. This test asserts that the pallet order is correct?

Reply:
Yes, so if the pallet order is incorrect (i.e. the `MessageQueue` pallet is before the `EthereumOutboundQueue` pallet in the `construct_runtime` macro), our message digest won't be found. I tested this locally, works nicely.